sympy/holonomic/holonomic.py

```diff
@@ -699,10 +878,39 @@ def _extend_y0(Holonomic, n):
         for i in range(n - a):
             sol = 0
             for a, b in zip(y1, list_red):
-                if not b.subs(x, Holonomic.x0).is_finite:
+                r = DMFsubs(b, Holonomic.x0)
+                if not r.is_finite:
                     return y0
-                sol += a * b.subs(x, Holonomic.x0)
+                sol += a * r
             y1.append(sol)
             list_red = _derivate_diff_eq(list_red)
 
         return y0 + y1[len(y0):]
+
+
+def DMFdiff(frac):
+    if not isinstance(frac, DMF):
+        return frac.diff()
+    K = frac.ring
+    R = K.dom.old_poly_ring(K.gens[0])
+    p = R(frac.num)
+    q = R(frac.den)
```

Comment:
There are methods for these: `K.numer(frac)` and `K.denom(frac)`. `R` is not needed. (There is also a method `R = K.get_ring()`.